### ISSUE TITLE
how to train the model to get the result same as the paper？

### ISSUE THREAD

**Issue by wyh196646** [Markin-Wang/XProNet#3]
Dear Author，I'm really enthusiastic about your fantastic work，I just train IU-Xray dataset use your code，and I get the best result is
test_BLEU_1    : 0.432083288799186
test_BLEU_2    : 0.27494955972033686
test_BLEU_3    : 0.19862342676532663
test_BLEU_4    : 0.1530441598906721
test_METEOR    : 0.17428250108359328
test_ROUGE_L   : 0.3539938538730797
and in training process ，the terminal show ‘Validation performance didn't improve for 50 epochs. Training stops.‘
In the same time, I juest test the model with the trained models you provide, I get the result same as the paper,I‘m really confused
,So,could you supply more details about how to train the best model?
Sincerely

**Comment by Markin-Wang** [comment]
> Dear Author，I'm really enthusiastic about your fantastic work，I just train IU-Xray dataset use your code，and I get the best result is test_BLEU_1 : 0.432083288799186 test_BLEU_2 : 0.27494955972033686 test_BLEU_3 : 0.19862342676532663 test_BLEU_4 : 0.1530441598906721 test_METEOR : 0.17428250108359328 test_ROUGE_L : 0.3539938538730797 and in training process ，the terminal show ‘Validation performance didn't improve for 50 epochs. Training stops.‘ In the same time, I juest test the model with the trained models you provide, I get the result same as the paper,I‘m really confused ,So,could you supply more details about how to train the best model? Sincerely

Hi, thanks for your interest.
Please train the model via the provided script: run_iu_xray.sh. I have tried my best to ensure the reproducibility of our work, e.g., fixed seed and use the determinstic algorithm. Hence, the results are exactly the same for each run in my workstation.
I would recommend you to set the same environment of ours.
In addition, as pointed by other repos, e.g.,[R2Gen](https://github.com/cuhksz-nlp/R2Gen/issues/5), the language model is highly sensitive to the hyper-parameters, and the IU X-Ray dataset is small, so the variance of the results may be larger.  You can try to tune some hyper-parameters.

**Comment by wyh196646** [comment]
Really appreciate for your clear reponse！

**Comment by njan-creative** [comment]
![R2GenResults](https://user-images.githubusercontent.com/65466327/203911275-f0a81b0e-a741-43e3-8475-123741f1600e.png)

I am also getting results similar to the values in this bug.

**Comment by LangWY** [comment]
Hello, I also have the same question. I wonder if you have achieved the results  on the IU-Xray dataset now.If successful, I hope you can tell me how you solved it, for example, which hyperparameter have been adjusted.

**Comment by Markin-Wang** [comment]
> Hello, I also have the same question. I wonder if you have achieved the results on the IU-Xray dataset now.If successful, I hope you can tell me how you solved it, for example, which hyperparameter have been adjusted.

Hi, thanks for your interest. I would recommend you to tune the learning rate first. Note that the language model is very sensitive to the learning rate, especially on the small dataset.

**Comment by LangWY** [comment]
Thank you for your answer. Your model is really effective, and I will try according to your suggestions.

**Comment by baoning1208** [comment]
> 谢谢你的回答。你的模型真的很有效，我会按照你的建议尝试。

This is indeed a great job, dear comrade, if you get good results, please let me know your hyperparameters
